Library/Homebrew/dev-cmd/bottle.rb

```diff
@@ -717,10 +717,7 @@ def merge_bottle_spec(old_keys, old_bottle_spec, new_bottle_hash)
 
   def old_checksums(formula, formula_ast, bottle_hash, args:)
     bottle_node = formula_ast.bottle_block
-    if bottle_node.nil?
-      odie "`--keep-old` was passed but there was no existing bottle block!" if args.keep_old?
-      return
-    end
+    return if bottle_node.nil? && args.keep_old?
     return [] unless args.keep_old?
```

Comment:
I think this will fix the tests:

```suggestion
    return if bottle_node.nil?
    return [] unless args.keep_old?
```